### DISCUSSION TITLE
LinkAce scalability and upper limits on the number of links and tags

### DISCUSSION THREAD

**Discussion by flamingjupiter** [Kovah/LinkAce#643] [Support]
Hi, as someone with a lot of bookmarks and no programming knowledge I'd like to know how many links LinkAce is able to handle. 

I figure this will depend on the specs of the server it is hosted, so, to give it a good guesstimate, 

Let's say I have a powerful rig, high end consumer level PC. I use it to run LinkAce. Would it be able to handle 100k links and 5k tags? Somewhere along those lines? Or that's way above it's capacity? 

And with handle I mean for it to function without becoming extremely slow and being to search links and/or tags without crazy response times. 

Thanks in advance for the help!

**Comment by Kovah** [comment]
Hi,
I have not tested LinkAce with so many links, lists or tags yet. However, even on a low-powered Raspberry Pi it should easily handle more than 1 million links and the same amount of lists or tags.

To give you some numbers to compare with: another project of mine running the same stack as LinkAce on a VPS with 2 cores and 4GB RAM has a table with more than 12 million entries. Running searches  takes 4-6 seconds which is quite reasonable, and those searches are not even optimized yet. So, there's plenty of room to scale.

**Comment by flamingjupiter** [comment]
I see, that's excellent to hear. Thank you for the reply and this amazing software you built!!
